### PR TITLE
LOD: Add `removeLevel()`.

### LINKED ISSUE #29074: Remove level in LOD?
### Description

Hi, are you interesting in a small feature for the LOD instance, that remove a specific level based on the distance?
I notice that currently there's none

I would like to take this one if you agree

### Solution

LOD.removeLevel(10) // 10 should be the distance, which is the level to delete

### Alternatives

Any other alternative, please feel free to comment here

### Additional context

As I'm working to add LOD on tresjs, I would like to have reactivity, to be able to modify the level or the hysteresis.
To do this, I should remove the current instance and create a new one

### PR DESCRIPTION
**Pull request by JaimeTorrealba**
Related issue: #29074

closes #29074

**Description**

Add the method removeLevel to LOD class

### CODE REVIEW

File: src/objects/LOD.js [hunk 1]
```diff
@@ -74,6 +74,23 @@ class LOD extends Object3D {
 
 	}
 
+	removeLevel( level ) {
```
Comment (Mugen87) [inline src/objects/LOD.js:77]: Now that I think about it: Can't you just directly remove the level from the `levels` array?

Do you mind providing a code snippet that shows how you get/create the `level` parameter before removing it from the LOD?

Comment (Mugen87) [inline src/objects/LOD.js:77]: Shouldn't the parameter be `distance`?

Comment (JaimeTorrealba) [reply src/objects/LOD.js:77]: Yep, it should be `distance`

File: src/objects/LOD.js [hunk 2]
```diff
@@ -74,6 +74,23 @@ class LOD extends Object3D {
...
 	}
 
+	removeLevel( distance ) {
+
+		const levels = this.levels;
+		levels.map( ( _level, index ) => {
+
+			if ( _level.distance === distance ) {
+
+				levels.splice( index, 1 );
```
Comment (Mugen87) [inline src/objects/LOD.js:84]: I'm not using `map()` that often so: Is it safe to remove an array instance inside the callback while processing the array? You couldn't do that with a normal for loop with ascending index.

Comment (JaimeTorrealba) [reply src/objects/LOD.js:84]: Yep there's no problem. I'm using the map just to loop over the levels.

Now if you prefer I can change that for a "for loop", depending on you, but in the end it is the same.

PS: I tested it in the current LOD example, but I didn't want to change it

Comment (John-Simth) [reply src/objects/LOD.js:84]: I think `const levels = this.levels.filter(_level => _level.distance !== distance)` maybe better.

Comment (Mugen87) [reply src/objects/LOD.js:84]: We mostly use `for` loop in the code base so for consistency reasons I vote for a reverse for loop.

Comment (JaimeTorrealba) [reply src/objects/LOD.js]: Done :)

File: src/objects/LOD.js [hunk 3]
```diff
@@ -77,15 +77,16 @@ class LOD extends Object3D {
 	removeLevel( distance ) {
 
 		const levels = this.levels;
-		levels.map( ( _level, index ) => {
+		for ( let i = 0; i < levels.length; i ++ ) {
```
Comment (Mugen87) [inline src/objects/LOD.js:80]: Can we do a reverse loop? In this case, it is no necessary to modify the index inside the loop.

Besides, would it be possible to early out when the level has been found? What is the use case for having more than on LOD for a  single distance value?

Comment (Mugen87) [reply src/objects/LOD.js:80]:
```js
for ( let i = levels.length - 1; i >= 0; -- i ) {
```

Comment (Mugen87) [reply src/objects/LOD.js:80]: But like I said before if there is no compelling use case for more than one LOD per level, the loop direction does not matter. You would return after the removal and that's it.

Comment (Mugen87) [reply src/objects/LOD.js:80]: I think the method should look like so:
```js
removeLevel( distance ) {

	const levels = this.levels;

	for ( let i = 0; i < levels.length; i ++ ) {

		if ( levels[ i ].distance === distance ) {

			const removedElements = levels.splice( i, 1 );
			this.remove( removedElements[ 0 ].object );

			return true;

		}

	}

	return false;

}
```
Would that work for you?

Comment (JaimeTorrealba) [reply src/objects/LOD.js:81]: The change is done and tested

**Pull request merged**